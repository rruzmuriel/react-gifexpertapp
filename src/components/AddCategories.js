import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategories = ({setCategories}) => {
    
    const [inputValue, setInputValue] = useState('')
    
    // Último valor actualizado que la persona escribió
    const handleInputChange = (e)=>{
        setInputValue(e.target.value);    
    }

    const handleSubmit = (e)=>{
        e.preventDefault();

        console.log('handleSubmit llamado', inputValue)

        if(inputValue.trim().length > 2){
            
            setCategories(((cats)=>[inputValue, ...cats ]));
            setInputValue('')           
        }
    }

    return (
        <form onSubmit = {handleSubmit}>
           
            <input 
                type="text"
                value={inputValue}
                onChange= {handleInputChange}
            />
        </form>
    )
}
AddCategories.propsTypes = {
    setCategories: PropTypes.func.isRequired
}
