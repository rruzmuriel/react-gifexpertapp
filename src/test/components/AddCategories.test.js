import React from 'react';
import '@testing-library/jest-dom'

import { shallow } from 'enzyme'
import { AddCategories } from '../../components/AddCategories'

describe('Pruebas en <<AddCategories />', () => {
    
    const setCategories= jest.fn();
    let wrapper =  shallow(<AddCategories setCategories={setCategories} />)

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper =  shallow(<AddCategories setCategories={setCategories} />)

    })

    test('Debe de mostrarse correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de cambiar la caja de texto', () => {

        const input = wrapper.find('input');
        const value = 'Hola Mundo'

        input.simulate('change', {target: {value} });

        expect( wrapper.find('p').text().trim() ).toBe(value);
        
    })
    
    test('No debe de postear la informaciÃ³n con submit', () => {

        wrapper.find('form').simulate('submit', { preventDefault(){} })
        
        expect( setCategories ).not.toHaveBeenCalled();
    })

    test('Debe de llamar setCategories y limpiar la caja de texto', () => {
        
        const value = 'Hola Mundo';

        // 1. Simular el inputChange
        wrapper.find('input').simulate('change', { target: {value} });

        // 2. Simular el submit
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        //3. setCategories se debe de haber llamaldo
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenLastCalledWith( expect.any(Function));

        // 4. el valor del input debe de estar ''
        expect( wrapper.find('input').prop('value')).toBe('');

    })
    
    
    
    
   
})
