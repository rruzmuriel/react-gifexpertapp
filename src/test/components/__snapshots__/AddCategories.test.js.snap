// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <<AddCategories /> Debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
